src/service/resize-observer.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {pushIfNotExist, removeItem} from '../utils/array';
+import {
+  registerServiceBuilder,
+  registerServiceBuilderInEmbedWin,
+} from '../service';
+import {rethrowAsync} from '../log';
+
+/** @typedef {function(../layout-rect.LayoutSizeDef)} */
+let ObserverCallbackDef;
+
+/**
+ * Pools resize observers. Eventually the hope is that the call sites can
+ * instantiate their own resize observers. But currently there are some
+ * performance issues and pooling is desired. For more info, see
+ * https://bugs.chromium.org/p/chromium/issues/detail?id=1157544
```

Comment:
Thanks for linking this bug! Very interesting results with a significant impact on how to structure our ResOb code.